stack-graphs/src/json.rs

```diff
@@ -348,27 +356,51 @@ impl<'a, T> InPaths<'a, T> {
 //-----------------------------------------------------------------------------
 // Paths
 
-impl Paths {
-    pub fn to_json_value(&mut self, graph: &StackGraph) -> Result<Value, JsonError> {
-        let paths = self.to_path_vec(graph);
-        Ok(serde_json::to_value(&InPaths(&paths, self, graph))?)
+impl<'a> Paths {
+    pub fn to_json<F>(&'a mut self, graph: &'a StackGraph, f: F) -> JsonPaths<'_, F>
+    where
+        F: Fn(&StackGraph, &Paths, &Path) -> bool,
```

Comment:
Can we make this `FnMut` instead of `Fn`?  That would provide more flexibility to callers, at the expense of ruling out ever being able to process paths in parallel in the future.

Reply:
It feels a little like mixing paradigms to me, but I see that this is also what `Iterator::filter` does, so I'll change this